File: frontend/src/api/premium.ts
import { client } from './client'
import type { User, PremiumStatus, PremiumProfileUpdate } from '../types'

export async function buyPremium(): Promise<User> {
  const res = await client.post('/premium/buy')
  return res.data
}

export async function getPremiumStatus(): Promise<PremiumStatus> {
  const res = await client.get('/premium/status')
  return res.data
}

export async function updatePremiumProfile(data: PremiumProfileUpdate): Promise<User> {
  const res = await client.patch('/premium/profile', data)
  return res.data
}

export async function disablePremium(): Promise<User> {
  const res = await client.post('/premium/disable')
  return res.data
}

File: routers/premium.py
from fastapi import APIRouter, Depends
from sqlalchemy.ext.asyncio import AsyncSession

from core.database import get_db
from core.dependencies import get_current_user
from models.user import User
from schemas.premium import PremiumProfileUpdate, PremiumStatusResponse
from schemas.user import UserResponse
from services.premium_service import buy_premium, update_premium_profile, disable_premium

router = APIRouter(prefix="/premium", tags=["Premium"])


@router.post("/buy", response_model=UserResponse)
async def buy(
    current_user: User = Depends(get_current_user),
    db: AsyncSession = Depends(get_db),
):
    # MVP: выдаём премиум сразу.
    return await buy_premium(db, current_user.id)


@router.get("/status", response_model=PremiumStatusResponse)
async def status(current_user: User = Depends(get_current_user)):
    # Текущий статус можно считать прямо из user.
    return current_user


@router.patch("/profile", response_model=UserResponse)
async def update_profile(
    data: PremiumProfileUpdate,
    current_user: User = Depends(get_current_user),
    db: AsyncSession = Depends(get_db),
):
    return await update_premium_profile(db, current_user.id, data)


@router.post("/disable", response_model=UserResponse)
async def disable(
    current_user: User = Depends(get_current_user),
    db: AsyncSession = Depends(get_db),
):
    return await disable_premium(db, current_user.id)

